app/templates/_widget.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference path="../../../tsd.ts.ts" />
+/// <amd-dependency path="dojo/text!./templates/<%=widgetName%>.html" name="template" />
+declare var template: any;
+
+//dojo
+import dojoDeclare = require("dojo/_base/declare");
+
+//dijit
+import WidgetBase = require("dijit/_WidgetBase");
+import TemplatedMixin = require("dijit/_TemplatedMixin"); 
+<% if(widgetsInTemplate) { %>import WidgetsInTemplateMixin = require("dijit/_WidgetsInTemplateMixin"); <% }%>
+
+interface WidgetInterface {
```

Comment:
Yes I could do a WidgetsInTemplate interface and a TemplatedWidget interface. However, within the scope of the scaffolder we wouldn't have access to a central repository to store them I believe? Also the reason I use the interface is because the type created by dojoDeclare is type any. Thus using 'this' anywhere within the widget doesn't bring up any of the intellisense. So I always cast this into self using the interface. The interface is therefore ends up unique to the class. I only ever add startup, placeAt, templateString and constructor which actually relate to the actual basewidget.

Maybe this is too specific to my way of working? But I've yet to see a better suggestion for developing widgets with TS. Maybe I should comment to explain this?